Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py

```diff
@@ -4389,6 +4466,366 @@ def get_indicator_dbot_object(indicator):
+    """ Retrieves list of vulnerabilities by software.
+        Args:
+            client: MsClient.
+            args: dict - arguments from CortexSOAR.
+        Returns:
+            A CommandResult list with a list of vulnerabilities by software.
+    """
+    results_list = []
+    software_id = str(args.get('id'))
+    headers = ['id', 'name', 'description', 'severity', 'cvssV3', 'publishedOn', 'updatedOn',
+               'exposedMachines', 'exploitVerified',
+               'publicExploit']
+    vulnerabilities_response = client.get_list_vulnerabilities_by_software(software_id)
+    vulnerabilities_response_value = vulnerabilities_response.get('value')
+    demisto.debug(vulnerabilities_response_value)
+    if vulnerabilities_response_value:
+        for cve in vulnerabilities_response_value:
+            cve_id = cve.get('id')
+            cve_indicator = Common.CVE(id=cve_id,
+                                       cvss=cve.get('cvssV3'),
+                                       description=cve.get('description'),
+                                       published=cve.get('publishedOn'),
+                                       modified=cve.get('updatedOn')
+                                       )
+            human_readable = tableToMarkdown(f'{INTEGRATION_NAME} vulnerability {cve_id} by software: {software_id}',
+                                             add_backslash_infront_of_underscore(cve), headers=headers, removeNull=True)
+            results_list.append(CommandResults(outputs_prefix='MicrosoftATP.SoftwareCVE',
+                                               outputs_key_field='id',
+                                               outputs=vulnerabilities_response_value,
+                                               readable_output=human_readable,
+                                               raw_response=vulnerabilities_response,
+                                               indicator=cve_indicator))
+    return results_list
+
+
+def create_filters_conjunction(filters_arg_list: list[str], name: str) -> str:
+    """ Create filter conjunction (added 'or' between args)
+        example output: id eq 'id1' or id eq 'id2'
+        Args:
+            filters_arg_list: list[str].
+            name: str.
+        Returns:
+            A str corresponding to the filter param in a qury.
+    """
+    query = ''
+    filters_arg_list = list(filter(None, filters_arg_list))
+    list_length = len(filters_arg_list)
+    if filters_arg_list:
+        for index, list_item in enumerate(filters_arg_list):
+            if index == list_length - 1 or list_length == 1:
+                query = f"{query}{name} eq '{list_item}'"
+            else:
+                query = f"{query}{name} eq '{list_item}' or "
+        demisto.debug(query)
+    return query
+
+
+def add_backslash_infront_of_underscore_list(markdown_data: Optional[list[dict]]) -> list[dict]:
```

Comment:
Why do we have 2 functions that does the same? 
Why not use add_backslash_infront_of_underscore for both?

Reply:
deleted one